### DISCUSSION TITLE
How to use runOnX callbacks in docker container?

### DISCUSSION THREAD

**Discussion by twvacek** [bluenviron/mediamtx#2930] [Questions] [answered]
### Question

I'm using the arm64 docker image. I need to add some packages that are required in the callback script. Is there a way?

**Comment by twvacek** [comment, marked as answer]
To answer my own question, the -rpi docker builds have sh, bash, and curl. This was adequate for me, but there appears to be a package manager as well.

**Reply by aler9** [reply]
If you need to add an executable to a Docker image, and i mean any docker image, not just the MediaMTX one, you can either build a custom image, based on the original image, and add any package you want by using the package manager of the base distro:

```
FROM mediamtx:latest-rpi
RUN apk add --no-cache ffmpeg
```

or alternatively you can build a brand new image and import the MediaMTX executable:

```
FROM mediamtx AS mediamtx
FROM ubuntu:22.04
RUN apt update && apt install -y ffmpeg
COPY --from=mediamtx /mediamtx /mediamtx
COPY --from=mediamtx /mediamtx.yml /mediamtx.yml
```

[a comment by an automated account is not shown]
